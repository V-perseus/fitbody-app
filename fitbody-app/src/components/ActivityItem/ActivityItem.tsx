import React from 'react'
import { View, Text, StyleSheet, Pressable, ViewProps, TextProps } from 'react-native'

import globals from '../../config/globals'
import { ActivitiesIconMap, ActivitiesIconMapType } from '../../config/svgs/dynamic/activitiesMap'

interface IActivityItemProps {
  iconKey: ActivitiesIconMapType
  name: string
  containerStyle?: ViewProps
  iconContainerStyle?: ViewProps
  iconColor?: string | null
  textStyle?: TextProps
  onPress?: () => void
  index: number
}
export const ActivityItem: React.FC<IActivityItemProps> = ({
  iconKey,
  name,
  containerStyle,
  iconContainerStyle,
  iconColor = null,
  textStyle,
  onPress,
  index,
}) => {
  const SVGIcon = ActivitiesIconMap?.[iconKey]
  const fillColor = iconColor ? iconColor : globals.styles.colors.colorWhite

  return (
    <Pressable onPress={onPress} disabled={!onPress}>
      <View style={[styles.tile, containerStyle]} testID={`activity_item_${index}`}>
        <View style={[styles.iconContainer, iconContainerStyle]}>
          {SVGIcon && <SVGIcon width={globals.window.width * 0.1} height={globals.window.width * 0.1} color={fillColor} />}
        </View>
        <Text style={[styles.tileText, textStyle]}>{name.toUpperCase().replace('\\N', '\n')}</Text>
      </View>
    </Pressable>
  )
}

const styles = StyleSheet.create({
  tile: {
    alignItems: 'center',
    height: 117,
    marginRight: 32,
  },
  iconContainer: {
    width: globals.window.width * 0.1546, // = 64 / 414
    height: globals.window.width * 0.1546,
    borderRadius: globals.window.width * 0.0773,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: globals.styles.colors.colorPink,
  },
  tileText: {
    marginTop: 6,
    fontFamily: globals.fonts.secondary.style.fontFamily,
    fontSize: 16,
    textAlign: 'center',
    color: globals.styles.colors.colorPink,
  },
})
