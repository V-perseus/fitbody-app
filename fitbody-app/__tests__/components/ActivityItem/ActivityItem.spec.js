import React from 'react'
import { render } from '@testing-library/react-native'
import { toHaveStyle } from '@testing-library/jest-native'
import renderer from 'react-test-renderer'

import { ActivityItem } from '../../../src/components/ActivityItem/ActivityItem'

const textStyle = {
  marginTop: 6,
  fontFamily: 'BebasNeue',
  fontSize: 16,
  textAlign: 'center',
}

describe('ActivityItem', () => {
  it('renders snapshot correctly', () => {
    const tree = renderer
      .create(
        <ActivityItem
          name={'exercised'}
          onPress={() => {}}
          iconColor={'#bada55'}
          // iconKey={'exercised'}
          // containerStyle={}
          // iconContainerStyle={}
          textStyle={textStyle}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the correct activity', () => {
    expect.extend({ toHaveStyle })
    const { queryByText } = render(
      <ActivityItem
        name={'exercised'}
        onPress={() => {}}
        iconColor={'#bada55'}
        // iconKey={'exercised'}
        // containerStyle={}
        // iconContainerStyle={}
        textStyle={textStyle}
      />,
    )
    expect(queryByText('EXERCISED')).not.toBeNull()
    expect(queryByText('EXERCISED')).toHaveStyle(textStyle)
  })
})
