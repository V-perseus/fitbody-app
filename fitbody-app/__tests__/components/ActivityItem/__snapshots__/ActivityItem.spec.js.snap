// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityItem renders snapshot correctly 1`] = `
<View
  accessibilityState={
    {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 117,
          "marginRight": 32,
        },
        undefined,
      ]
    }
    testID="activity_item_undefined"
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#fa8dd5",
            "borderRadius": 57.974999999999994,
            "height": 115.94999999999999,
            "justifyContent": "center",
            "width": 115.94999999999999,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#fa8dd5",
            "fontFamily": "BebasNeue",
            "fontSize": 16,
            "marginTop": 6,
            "textAlign": "center",
          },
          {
            "fontFamily": "BebasNeue",
            "fontSize": 16,
            "marginTop": 6,
            "textAlign": "center",
          },
        ]
      }
    >
      EXERCISED
    </Text>
  </View>
</View>
`;
